solr/core/src/java/org/apache/solr/search/QueryResultKey.java

```diff
@@ -65,6 +70,7 @@ public QueryResultKey(Query query, List<Query> filters, Sort sort, int nc_flags)
       h = h*29 + sf.hashCode();
       ramSfields += BASE_SF_RAM_BYTES_USED + RamUsageEstimator.sizeOfObject(sf.getField());
     }
+    h = h*31 + minExactHits;
```

Comment:
this is going to overflow in the default case, is that ok?

Reply:
ah! Good question. I guess yes?

Reply:
Added a test